packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,395 @@
+  readonly path?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefix?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines gRPC gateway route rewrite that can be defined in addition to host name.
+ */
+export abstract class HttpGatewayRoutePathOrPrefixRewrite {
+  /**
+   * The path to rewrite.
+   *
+   * @param exact The exact path to rewrite.
+   */
+  public static path(exact: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePathRewriteImpl(exact);
+  }
+
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   *
+   * @param isDefault Boolean to select to enable or disable the default prefix. Default value is enabled.
+   */
+  public static defaultPrefix(isDefault?: boolean): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ defaultPrefix: isDefault == false ? 'DISABLED' : 'ENABLED' });
+  }
+
+  /**
+   * Replace the incoming route prefix when rewritten.
+   *
+   * @param value The value used to replace the incoming route prefix when rewritten.
+   */
+  public static customPrefix(value: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ value: value } );
+  }
+
+  /**
+   * Return HTTP gateway route rewrite configuration.
    */
-  readonly prefixPath: string;
+  abstract bind(scope: Construct): HttpGatewayRouteRewriteConfig;
+}
+
+class HttpGatewayRoutePathRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly exact: string,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      path: {
+        exact: this.exact,
+      },
+    };
+  }
+}
+
+class HttpGatewayRoutePrefixRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly prefixRewrite: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      prefix: this.prefixRewrite,
+    };
+  }
+}
+
+/**
+ * The criterion for determining a rewrite for this HTTP GatewayRoute.
+ */
+export interface HttpGatewayRouteRewrite {
+  /**
+   * The default target host name to write to.
+   * Boolean is used to select either enable or disable.
+   *
+   * @default true
+   */
+  readonly defaultHostname?: boolean;
+
+  /**
+   * The path to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix or path is rewritten to '/'.
+   */
+  readonly pathOrPrefix?: HttpGatewayRoutePathOrPrefixRewrite,
 }
 
 /**
- * The criterion for determining a request match for this GatewayRoute
+ * Defines gRPC gateway route rewrite
  */
-export interface GrpcGatewayRouteMatch {
+export interface GrpcGatewayRouteRewrite {
```

Comment:
We don't do single-property interfaces like this, there's really no reason.

Remove this interface, and add this new property directly to `GrpcGatewayRouteSpecOptions` (with possibly a slightly different name).

Reply:
I created the interface because it is possible to add more options in the future. However, it is not planned as of now.

Will remove this interface.